cvmfs/server/cvmfs_server_ingest.sh

```diff
@@ -210,6 +228,15 @@ cvmfs_server_ingest() {
     ingest_command="$ingest_command -+stats"
   fi
 
+  local upstream_storage=$CVMFS_UPSTREAM_STORAGE
+  local upstream_type=$(get_upstream_type $upstream_storage)
+  gw_key_file=/etc/cvmfs/keys/${name}.gw
+
+  if [ x"$upstream_type" = xgw ]; then
+    ingest_command="$ingest_command -H $gw_key_file -P ${spool_dir}/session_token"
```

Comment:
You are passing the gw key and the session token. Below, I don't see how these are used.

Reply:
Never mind this, you explained below.